File: src/main/java/fm/comunas/ddd/oe/model/ObjTenantRepository.java

package fm.comunas.ddd.oe.model;

import fm.comunas.ddd.obj.model.ObjRepository;
import fm.comunas.ddd.oe.model.db.tables.records.ObjTenantVRecord;

import java.util.Optional;

public interface ObjTenantRepository extends ObjRepository<ObjTenant, ObjTenantVRecord> {

	/**
	 * Lookup tenant with given id (in global session)
	 */
	Optional<ObjTenant> get(Integer id);

	/**
	 * Lookup Tenant by extl key
	 */
	Optional<ObjTenant> getByExtlKey(String extlKey);

}

File: fm-dddrive/src/test/java/io/dddrive/domain/oe/model/ObjTenantRepository.kt
package io.dddrive.domain.oe.model

import io.dddrive.obj.model.ObjRepository
import io.dddrive.oe.model.ObjTenant
import java.util.*

interface ObjTenantRepository : ObjRepository<ObjTenant> {

	fun getByKey(key: String): Optional<ObjTenant>

	companion object {

		const val KERNEL_TENANT_KEY: String = "k"
	}

}
